csharpcore/GildedRose/GildedRose.cs

```diff
@@ -1,89 +1,94 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace GildedRoseKata
 {
-    public class GildedRose
+  public class GildedRose
+  {
+    static IList<Item> Items;
+    public const string BACKSTAGE_PASS_NAME = "Backstage passes to a TAFKAL80ETC concert";
+    public const string SULFURAS_NAME = "Sulfuras, Hand of Ragnaros";
+    public const string AGED_BRIE_NAME = "Aged Brie";
+    private static List<string> specialProductNames = new() { BACKSTAGE_PASS_NAME, SULFURAS_NAME, AGED_BRIE_NAME };
+
+    public static void SetInventory(IList<Item> items)
+    {
+      GildedRose.Items = items;
+    }
+
+    // TODO: Deprecate this. keeping the constructor around for (imaginary) legacy code
+    public GildedRose(IList<Item> items)
+    {
+      GildedRose.Items = items;
+    }
+
+    public static void UpdateQuality(IList<Item> items)
+    {
+      SetInventory(items);
+      UpdateQuality();
+    }
+
+    public static void UpdateQuality()
+    {
+      foreach (Item i in Items.Where(i => i.Name == BACKSTAGE_PASS_NAME))
```

Comment:
For performance reasons, consider flipping this to

```
foreach(Item i in Items) {
  switch (i.Name) {
    case BACKSTAGE_PASS_NAME:
      updateBackstagePass(i);
      break;
    // etc
    default:
      updateOther(i);
      break;
  }
}
```
Doing so will significantly reduce the amount of string comparisons (`Items[x]` will be compared to each special name 6 times), and will reduce allocations (behind the scenes, `foreach` allocates an enumerator).

Fun fact: this `foreach` will get translated to something like

```
var enumerator is whereResult.GetEnumerator();
try {
  while (enumerator.MoveNext()) {
    i = enumerator.Current;
    updateBackstagePass(i);
  }
}
finally {
  enumerator.Dispose();
}
```